### DISCUSSION TITLE
GLFW+MoltenVK cross-compilation on Linux

### DISCUSSION THREAD

**Discussion by tomas-mraz** [KhronosGroup/MoltenVK#1702] [Q&A]
Hi, I am making a cross-compilation Golang application on Linux with GLFW and MoltenVK for macOS.
I have pre-requisition "the **MoltenVK.framework** needs to be installed in the /Library/Frameworks folder for the build to find it".
It seems script "./fetchDependencies" heavily depends on Apple Xcode SDK.
I end up on
```
Building external libraries for platform macOS and destination macOS
./fetchDependencies: line 383: xcodebuild: command not found
```
I can modify the destination OS in the script but it does not change the fact I need to run "xcodebuild".
Does have somebody any hints?

**Comment by billhollings** [comment]
If I understand what you are trying to do, you want to link MoltenVK into an application build (cross-build) on Linux? But that app would then actually run on an Apple platform, presumably?

Is there some reason why you need to build MoltenVK itself on Linux? Could you use a MoltenVK binary that was built under macOS (or a macOS VM on Linux)?

**Reply by Gcenx** [reply]
The reason for trying to build MoltenVK from source would be due to the LunarG VulkanSDK packages have move d to a 7z based installer package that’s packaged inside an APFS formatted dmg image making it rather irritating to extract.

**Reply by richard-lunarg** [reply]
Out of curiosity, is it the fact that you have to run the installation program (it can be run from the command line) the irritation, or the fact that you need to mount a .dmg first? If we distributed the installer as a .zip instead, would this be better for you?

**Reply by Gcenx** [reply]
@richard-lunarg the major hurdle is being packaged inside a dmg.

The installer would be the second annoyance as that requires macOS 10.13 to run. This can be avoided by extracting the contents of InstallVulkan.app/Contents/Resources/installer.dat using 7z but still rather annoying.

> If we distributed the installer as a .zip instead, would this be better for you?

This would be preferable as it would allow using the latest version of MoltenVK even on OS X 10.11 (this doesn’t support APFS dmg images) or via osxcross.

If this was instead packaged within a .zip archive I know Macports would be rather happy.

**Reply by richard-lunarg** [reply]
Thanks @Gcenx for the feedback. I've created an internal issue so we can discuss this at LunarG.

**Comment by Gcenx** [comment]
Your version of osxcross is rather old as the newer ones do include an xcodebuild wrapper however it lightly still won’t help building MoltenVK from source.

You could instead use one of the stock builds from my mirror https://github.com/The-Wineskin-Project/MoltenVK/releases all built via GitHub actions.
